ipa/ipa_utils.py

```diff
@@ -457,6 +457,22 @@ def parse_test_name(name):
     return '::'.join(filter(None, [test_file, test_class, test_case]))
 
 
+def put_file(client, source_file, destination_file):
+    """
+    Copy file to instance using Paramiko client connection.
+    """
+    try:
+        sftp_client = client.open_sftp()
+        sftp_client.put(source_file, destination_file)
```

Comment:
Are you planning to encode the "default location" such as "/tmp/injected" at the next level up? WHich is OK

Reply:
Yes, I think any directories should be provided to the utility method. The base provider method which will process injection params should provide this as it will be the same for handling each type of injection.